dojson/contrib/marc21/utils.py

```diff
@@ -126,7 +126,22 @@ def split_blob(blob):
     raise StopIteration()
 
 
+def split_stream(stream):
+    """Split the stream using <record.*?>.*?</record> as pattern."""
+    buf = []
+    for row in stream:
+        index = row.rfind("</record>")
```

Comment:
> this can work for a specific and well-formatted MARC XML. we should better investigate solution with `xml.sax.xmlreader`.

This is true, but the proposed implementation is as correct as the previous one but without the memory issue. Therefore it's quite an important improvement, albeit not the definitive one.